app/models/calculator/weight.rb

```diff
@@ -57,8 +57,13 @@ def quantity_implied_in_final_weight_volume(line_item)
       (1.0 * line_item.final_weight_volume / line_item.variant.unit_value).round(3)
     end
 
+    def use_final_weight_volume?(line_item)
+      final_weight_volume_present?(line_item) && !line_item.variant.unit_value.zero?
```

Comment:
I dont think this is correct. If variant_unit is weight, you dont need unit_value to use final_weight_volume.
I think the previous version of the code is better where we just change quantity_implied_in_final_weight_volume